easybuild/easyconfigs/a/ADIOS2/ADIOS2-2.10.2-gompi-2024a.eb

```
@@ -0,0 +1,54 @@
+easyblock = "CMakeMake"
+
+name = "ADIOS2"
+version = "2.10.2"
+
+homepage = 'https://github.com/ornladios/ADIOS2/'
+description = """This is ADIOS2: The Adaptable Input/Output (I/O) System.
+ADIOS2 is developed as part of the United States Department of Energy's Exascale Computing
+Project. It is a framework for scientific data I/O to publish and subscribe to data when
+and where required.
+ADIOS2 transports data as groups of self-describing variables and attributes across different
+media types (such as files, wide-area-networks, and remote direct memory access) using a
+common application programming interface for all transport modes. ADIOS2 can be used on
+supercomputers, cloud systems, and personal computers. """
+
+toolchain = {'name': 'gompi', 'version': '2024a'}
+
+source_urls = ['https://github.com/ornladios/ADIOS2/archive/refs/tags/']
+sources = ['v%(version)s.tar.gz']
+checksums = ['14cf0bcd94772194bce0f2c0e74dba187965d1cffd12d45f801c32929158579e']
+
+builddependencies = [
+    ('CMake', '3.29.3'),
+    ('Python', '3.12.3'),
+]
+
+dependencies = [
+    ('HDF5', '1.14.5'),
+    ('c-blosc', '1.21.6'),
+    ('libfabric', '1.21.0'),
+]
+
+configopts = ' '.join([
+    "-DBUILD_TESTING=ON",
```

```suggestion
    "-DBUILD_TESTING=ON",
    "-DADIOS2_RUN_MPI_MPMD_TESTS=OFF",
```

Not an expert on the tool... we could turn the MPI tests off like this, but maybe it should be investigated if the failures are actually significant or if it is just about properly pre-configuring the env?
https://adios2.readthedocs.io/en/v2.10.2/advanced/ecp_hardware.html

---

Tested locally and my errors go from

```
	945 - Engine.BP.BPJoinedArray.MultiBlock.BP4.MPI (Failed)
	947 - Engine.BP.BPJoinedArray.MultiBlock.BP5.MPI (Failed)
	1489 - Bindings.Fortran.BPWriteReadHeatMap6D.MPI (Failed)
	1581 - Utils.IOTest.Pipe2.SSC (Failed)
	1582 - Utils.IOTest.Pipe2.SSC.Dump (Not Run)
	1583 - Utils.IOTest.Pipe2.SSC.Validate (Not Run)
```

to 
```
The following tests FAILED:
	945 - Engine.BP.BPJoinedArray.MultiBlock.BP4.MPI (Failed)
	947 - Engine.BP.BPJoinedArray.MultiBlock.BP5.MPI (Failed)
	1474 - Bindings.Fortran.BPWriteReadHeatMap6D.MPI (Failed)
```

so this will probably also not be enough for the bot